src/envoy/auth/jwt.cc

```diff
@@ -90,19 +102,49 @@ const uint8_t *CastToUChar(const std::string &str) {
   return reinterpret_cast<const uint8_t *>(str.c_str());
 }
 
-bssl::UniquePtr<EVP_PKEY> EvpPkeyFromStr(const std::string &pkey_pem) {
-  std::string pkey_der = Base64::decode(pkey_pem);
-  bssl::UniquePtr<RSA> rsa(
-      RSA_public_key_from_bytes(CastToUChar(pkey_der), pkey_der.length()));
+bssl::UniquePtr<EVP_PKEY> EvpPkeyFromRsa(const bssl::UniquePtr<RSA> &rsa) {
   if (!rsa) {
     return nullptr;
   }
   bssl::UniquePtr<EVP_PKEY> key(EVP_PKEY_new());
   EVP_PKEY_set1_RSA(key.get(), rsa.get());
-
   return key;
 }
 
+bssl::UniquePtr<EVP_PKEY> EvpPkeyFromStr(const std::string &pkey_pem) {
+  std::string pkey_der = Base64::decode(pkey_pem);
+  return EvpPkeyFromRsa(bssl::UniquePtr<RSA>(
+      RSA_public_key_from_bytes(CastToUChar(pkey_der), pkey_der.length())));
+}
+
+BIGNUM *BigNumFromBase64UrlString(const std::string &s) {
+  std::string s_decoded = Base64UrlDecode(s);
+  if (s_decoded == "") {
+    return nullptr;
+  }
+  return BN_bin2bn(CastToUChar(s_decoded), s_decoded.length(), NULL);
```

Comment:
it is fine either, you can also

return bssl::UniquePtr and release it when assigns to rsa->n